.clang-format

```diff
@@ -0,0 +1,456 @@
+
+# ConstructorInitializerAllOnOneLineOrOnePerLine (Boolean) clang-format 3.7
+ConstructorInitializerAllOnOneLineOrOnePerLine: true
+
+# ConstructorInitializerIndentWidth (Unsigned) clang-format 3.7
+ConstructorInitializerIndentWidth: 4
+
+# ContinuationIndentWidth (Unsigned) clang-format 3.7
+ContinuationIndentWidth: 4
+
+# Cpp11BracedListStyle (Boolean) clang-format 3.4
+Cpp11BracedListStyle: true
+
+# DeriveLineEnding (Boolean) clang-format 10
+DeriveLineEnding: true
+
+# DerivePointerAlignment (Boolean) clang-format 3.7
+DerivePointerAlignment: false
+
+# DisableFormat (Boolean) clang-format 3.7
+DisableFormat:   false
+
+# EmptyLineAfterAccessModifier (EmptyLineAfterAccessModifierStyle) clang-format 13
+
+# EmptyLineBeforeAccessModifier (EmptyLineBeforeAccessModifierStyle) clang-format 12
+
+# ExperimentalAutoDetectBinPacking (Boolean) clang-format 3.7
+
+# FixNamespaceComments (Boolean) clang-format 5
+FixNamespaceComments: true
+
+# ForEachMacros (List of Strings) clang-format 3.7
+ForEachMacros:
+  - GEM5_FOR_EACH_IN_PACK
+
+# IfMacros (List of Strings) clang-format 13
+
+# IncludeBlocks (IncludeBlocksStyle) clang-format 6
+IncludeBlocks:   Preserve
+
+# IncludeCategories (List of IncludeCategories) clang-format 3.8
+IncludeCategories:
+  - Regex:           '^<.*\.h>'
+    Priority:        1
+    SortPriority:    0
+  - Regex:           '^<.*'
+    Priority:        2
+    SortPriority:    0
+  - Regex:           '.*'
+    Priority:        3
+    SortPriority:    0
+
+# IncludeIsMainRegex (String) clang-format 3.9
+IncludeIsMainRegex: '([-_](test|unittest))?$'
+
+# IncludeIsMainSourceRegex (String) clang-format 10
+IncludeIsMainSourceRegex: ''
+
+# IndentAccessModifiers (Boolean) clang-format 13
+
+# IndentCaseBlocks (Boolean) clang-format 11
+
+# IndentCaseLabels (Boolean) clang-format 3.3
+IndentCaseLabels: false
+
+# IndentExternBlock (IndentExternBlockStyle) clang-format 11
+
+# IndentGotoLabels (Boolean) clang-format 10
+IndentGotoLabels: true
+
+# IndentPPDirectives (PPDirectiveIndentStyle) clang-format 6
+IndentPPDirectives: AfterHash
+
+# IndentRequiresClause (Boolean) clang-format 15
+
+# IndentWidth (Unsigned) clang-format 3.7
+IndentWidth:     4
+
+# IndentWrappedFunctionNames (Boolean) clang-format 3.7
+IndentWrappedFunctionNames: false
+
+# InsertBraces (Boolean) clang-format 15
+
+# InsertNewlineAtEOF (Boolean) clang-format 16
+
+# InsertTrailingCommas (TrailingCommaStyle) clang-format 11
+
+# IntegerLiteralSeparator (IntegerLiteralSeparatorStyle) clang-format 16
+
+# JavaImportGroups (List of Strings) clang-format 8
+
+# JavaScriptQuotes (JavaScriptQuoteStyle) clang-format 3.9
+
+# JavaScriptWrapImports (Boolean) clang-format 3.9
+
+# KeepEmptyLinesAtTheStartOfBlocks (Boolean) clang-format 3.7
+KeepEmptyLinesAtTheStartOfBlocks: false
+
+# LambdaBodyIndentation (LambdaBodyIndentationKind) clang-format 13
+
+# LineEnding (LineEndingStyle) clang-format 16
+
+# MacroBlockBegin (String) clang-format 3.7
+MacroBlockBegin: "^\
```

Comment:
I suspect these regular expressions aren't matching what you expect due to operator precedence. That's probably why you had to rename EndBitUnion. I got this to work as expected:

```
MacroBlockBegin: "^(\
BitUnion(8|16|32|64)|\
SubBitUnion\
)$"

MacroBlockEnd: "^(\
EndBitUnion|\
EndSubBitUnion\
)$"
```
With the change above, you should be able to avoid renaming EndBitUnion.

Reply:
Fixed, thanks!